### PR TITLE
New: Add support for offset-based pagination

### PR DESCRIPTION
**Pull request by jstoffan**
pagination1

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #547

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/api/Folder.js [hunk 1]
```diff
@@ -105,50 +102,38 @@ class Folder extends Item {
...
         const {
             entries,
+            offset,
             total_count,
         }: FlattenedBoxItemCollection = item_collection;
         if (!Array.isArray(entries) || typeof total_count !== 'number') {
             throw getBadItemError();
         }
 
-        // Total count may be more than the actual number of entries, so don't rely
```
Comment (jeremypress) [inline src/api/Folder.js]: 👍

File: src/components/ContentExplorer/ContentExplorer.js [hunk 2]
```diff
@@ -1325,6 +1344,14 @@ class ContentExplorer extends Component<Props, State> {
                             onItemPreview={this.preview}
                             onSortChange={this.sort}
                         />
+                        <Footer>
+                            <Pagination
```
Comment (jeremypress) [inline src/components/ContentExplorer/ContentExplorer.js]: Will we always render pagination in the explorer? In webapp we don't show anything if the entries fit on one page.

Comment (jstoffan) [reply src/components/ContentExplorer/ContentExplorer.js]: The pagination controls aren't rendered if there's only one page and the footer is hidden via the `:empty` pseudo-class in that case.

File: src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -38,7 +38,8 @@ const Pagination = ({
     const pageCount = Math.ceil(totalCount / pageSize);
     if (pageCount <= 1) return null;
 
-    const pageNumber = Math.floor(offset / pageSize) + 1;
+    const pageNum = Math.floor(offset / pageSize) + 1;
+    const pageNumber = pageNum > 0 ? Math.min(pageCount, pageNum) : 1;
```
Comment (jeremypress) [inline src/components/Pagination/Pagination.js]: to me these names mean the same thing

**Comment by jstoffan** [comment]
Fixes https://github.com/box/box-ui-elements/issues/178.

**Pull request merged**